src/datasources/targeted-messaging/targeted-messaging.datasource.ts

```diff
@@ -200,9 +202,13 @@ export class TargetedMessagingDatasource
       outreachId: args.targetedSafe.outreachId,
       targetedSafeId: args.targetedSafe.id,
       signerAddress: args.signerAddress,
-      completionDate: new Date(submission.completion_date),
-      created_at: new Date(submission.created_at),
-      updated_at: new Date(submission.updated_at),
+      completionDate: this.parseDate(submission.completion_date),
+      created_at: this.parseDate(submission.created_at),
+      updated_at: this.parseDate(submission.updated_at),
     };
   }
+
+  private parseDate(date: Date | string): Date {
```

Comment:
Conditionally parsing the value to `Date` (if it's not a `Date` already) is still needed, because the values that come from the cache are ISO strings instead of `Date` objects. This was detected by this kind of tests: https://github.com/safe-global/safe-client-gateway/pull/2010#discussion_r1802848525

(cc. @iamacook)